src/Config/Type.hs

```diff
@@ -106,11 +106,27 @@ data Restrict = Restrict
     ,restrictWithin :: [(String, String)]
     } deriving Show
 
+data SmellType = SmellLongFunctions | SmellLongTypeLists | SmellManyArgFunctions | SmellManyImports
+  deriving (Show,Eq,Ord)
+
+getSmellType :: String -> Maybe SmellType
+getSmellType "long functions" = Just SmellLongFunctions
+getSmellType "long type lists" = Just SmellLongTypeLists
+getSmellType "many arg functions" = Just SmellManyArgFunctions
+getSmellType "many imports" = Just SmellManyImports
+getSmellType _ = Nothing
+
+data Smell = Smell
+    {smellType :: SmellType
+    ,smellLimit :: Int
+    } deriving Show
+
 data Setting
     = SettingClassify Classify
     | SettingMatchExp HintRule
     | SettingRestrict Restrict
     | SettingArgument String -- ^ Extra command-line argument
+    | SettingSmell Smell
```

Comment:
I'd be tempted to just inline `Smell` here, so `SettingSmell SemllType Int`. I'm not sure how much the `Smell` data type adds.